### PR TITLE
Add golangci-lint for BFF linting

### PR DESCRIPTION
**Pull request by Griffin-Sullivan**
## Description
Adding golangci-lint for BFF linting. This combines multiple linters to run against the BFF. I've also added a GitHub Action to run only against the BFF directory and a contributing doc with steps on how to install and run the linter locally.

## How Has This Been Tested?
Ran locally and added GHA

## Merge criteria:

- All the commits have been [_signed-off_](https://github.com/kubeflow/community/tree/master/dco-signoff-hook#signing-off-commits)  (To pass the `DCO` check)

- [x] The commits have meaningful messages; the author will squash them after approval or in case of manual merges will ask to merge with squash.
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has manually tested the changes and verified that the changes work.
- [x] Code changes follow the [kubeflow contribution guidelines](https://www.kubeflow.org/docs/about/contributing/).

If you have UI changes

- [ ] The developer has added tests or explained why testing cannot be added.
- [ ] Included any necessary screenshots or gifs if it was a UI change.
- [ ] Verify that UI/UX changes conform the UX guidelines for Kubeflow.

### CODE REVIEW

File: clients/ui/bff/internal/data/model_version_test.go [hunk 1]
```diff
@@ -2,17 +2,18 @@ package data
...
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/mock"
-	"net/http"
-	"net/url"
-	"testing"
 )
 
 func TestGetModelVersion(t *testing.T) {
-	gofakeit.Seed(0)
+	_ = gofakeit.Seed(0)
```
Comment (Griffin-Sullivan) [inline clients/ui/bff/internal/data/model_version_test.go:16]: The errcheck linter really didn't like that we weren't checking errors on these statements. This is an ugly hack so maybe I should try to configure it to ignore our test files for this linter.

File: clients/ui/bff/docs/contributing.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+```shell
+brew install golangci-lint
+```
+
+and to run it simply use:
+
+```shell
+cd clients/ui/bff
+golangci-lint run
```
Comment (ederign) [inline clients/ui/bff/docs/contributing.md:16]: Shall we add a target on makefile?

Similar to https://github.com/kubeflow/notebooks/blob/notebooks-v2/workspaces/backend/Makefile#L49-L56

File: clients/ui/bff/internal/api/errors.go [hunk 3]
```diff
@@ -91,18 +91,19 @@ func (app *App) methodNotAllowedResponse(w http.ResponseWriter, r *http.Request)
...
-	httpError := &integrations.HTTPError{
-		StatusCode: http.StatusUnprocessableEntity,
-		ErrorResponse: integrations.ErrorResponse{
-			Code:    strconv.Itoa(http.StatusUnprocessableEntity),
-			Message: string(message),
-		},
-	}
-	app.errorResponse(w, r, httpError)
-}
+// TODO: Uncomment when we implement proper error handling
```
Comment (ederign) [inline clients/ui/bff/internal/api/errors.go:94]: Why did we have to comment on this? To please the linter?

File: clients/ui/bff/internal/data/model_version_test.go (hunk 1 above)
Comment (ederign) [reply clients/ui/bff/internal/data/model_version_test.go:16]: For those cases https://golangci-lint.run/usage/false-positives/#nolint-directive also works.

**Review by ederign — CHANGES_REQUESTED** [review]
@Griffin-Sullivan could you please also add the targets for makefile?

File: clients/ui/bff/internal/api/errors.go (hunk 3 above)
Comment (Griffin-Sullivan) [reply clients/ui/bff/internal/api/errors.go:94]: Yeah it was an unused function so the linter was upset. We've been doing similar commenting out in the frontend so I figured to do the same here

File: clients/ui/bff/docs/contributing.md (hunk 2 above)
Comment (Griffin-Sullivan) [reply clients/ui/bff/docs/contributing.md:16]: can do

File: clients/ui/bff/internal/api/errors.go (hunk 3 above)
Comment (Griffin-Sullivan) [reply clients/ui/bff/internal/api/errors.go]: Decided to change this so that we use `//nolint`

**Review by ederign — APPROVED** [review]

**Comment by ederign** [comment]
@tarilabs, As this touches an action, we need your powers here. We hardcoded the version to have reproducibility.

**Comment by alexcreasy** [comment]
@Griffin-Sullivan looks good! 

Out of interest is there any kind of config file where we can turn things on and off a bit like ESLint?

**Review by tarilabs — APPROVED** [review]
- https://github.com/kubeflow/model-registry/pull/433#pullrequestreview-2339941217

/approve

**Pull request merged**